workers/unity/Assets/Gdk/Core/Collections/HandleProvider.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+
+namespace Assets.Gdk.Core.Collections
+{
+    internal static class HandleProvider
```

Comment:
This might end up being a fairly common name if we aren't careful. Might be worth name it something more specific.

Reply:
That doesn't really help. If this is a global then it should give a lot of context in it's name, and not be something that could clash with anything else.